arches/app/media/js/viewmodels/workflow.js

```diff
@@ -440,6 +434,16 @@ define([
             });
 
             window.location.assign(self.quitUrl);
+        }
+
+        this.finishWorkflow = function() {
+            if (self.activeStep().hasUnsavedData()) {
+                self.activeStep().save().then(function() {
+                    self.markWorkflowComplete();
+                });
+                return;
+            }
+            self.markWorkflowComplete();
```

Comment:
nit ( since we're refactoring ): replacing the `return` statement with `if/else` logic is slightly more readable.